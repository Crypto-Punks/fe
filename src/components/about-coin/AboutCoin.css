.AboutCoin {
  position: absolute;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  text-align: center;
  height: 100vh;
  width: 100vw;
  top: 88em;
  

    a {
      margin: 1em;
      text-decoration: none;
      color: white;
    }

  p {
    margin-top: 3%;
    margin-bottom: 10%;
    margin-left: 15%;
    margin-right: 15%;
    color: white;
    font-family: abolition, sans-serif;
    font-weight: 300;
    font-style: normal;
    letter-spacing: 3px;
  }

  .Stats {
    background-color: rgba(101, 134, 137, 0.845);
    text-align: center;
    margin: 3%;
    width: 60%;
    height: 15%;
    padding: 2%;
    border-radius: 1em;

    p {
      margin: 0;
    }
  }
}

@media (max-width: 500px) {

  .AboutCoin {
    top: 28em;
    margin-left: 8%;
    margin-right: 0;

    p {
      font-size: 10pt;
      text-align: center;
      margin-bottom: 15%;
      width: 90%;
      margin-left: 0;
      margin-right: 0;
    }

    .Stats {
      width: 90%;
      height: 20%;

      p {
        text-align: left;
      }
    }

  }
}