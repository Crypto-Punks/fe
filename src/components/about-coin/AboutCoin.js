import React from 'react';
import PropTypes from 'prop-types';
import styles from './AboutCoin.css';

//todo - maybe add info about volume, market cap, etc.

const AboutCoin = ({ website, description }) => {
  return (
    <div className={styles.AboutCoin}>
      <a target='_blank' rel="noopener noreferrer" href={website}>Coins website</a>
      <p>{description}</p>
    </div>
  );
};

AboutCoin.propTypes = {
  website: PropTypes.string,
  description: PropTypes.string
};

export default AboutCoin;
