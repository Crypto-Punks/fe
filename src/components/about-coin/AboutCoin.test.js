import React from 'react';
import { shallow } from 'enzyme';
import AboutCoin from './AboutCoin';

const data = {
  id: 'bitcoin',
  name: 'Bitcoin',
  price: '7231.3856839244778371',
  supply: '18098137.0000000000000000',
  maxSupply: '21000000.0000000000000000',
  marketCapUsd: '130874608807.5038975492994827',
  volumeUsd24Hr: '3535338669.2445264736224412',
  changePercent24Hr: '0.5591885701135556',
  logo: 'https://s2.coinmarketcap.com/static/img/coins/64x64/1.png',
  website: 'https://bitcoin.org/',
  description: 'Bitcoin (BTC) is a consensus network that enables a new payment system and a completely digital currency. Powered by its users, it is a peer to peer payment network that requires no central authority to operate. On October 31st, 2008, an individual or group of individuals operating under the pseudonym "Satoshi Nakamoto" published the Bitcoin Whitepaper and described it as: "a purely peer-to-peer version of electronic cash, which would allow online payments to be sent directly from one party to a...'
};

describe('AboutCoin component', () => {
  it('renders AboutCoin', () => {
    const wrapper = shallow(<AboutCoin { ...data} />);
    expect(wrapper).toMatchSnapshot();
  });
});
