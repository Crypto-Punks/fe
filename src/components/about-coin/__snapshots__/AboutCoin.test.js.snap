// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutCoin component renders AboutCoin 1`] = `
<div
  className="AboutCoin"
>
  <a
    href="https://bitcoin.org/"
    rel="noopener noreferrer"
    target="_blank"
  >
    Coins website
  </a>
  <div
    className="Stats"
  >
    <p>
      Supply: 
      18098137.0000000000000000
    </p>
    <p>
      Max Supply: 
      21000000.0000000000000000
    </p>
    <p>
      Market Cap in USD: 
      130874608807.5038975492994827
    </p>
    <p>
      Volume Traded in last 24 Hours in USD: 
      3535338669.2445264736224412
    </p>
    <p>
      Change Percent in last 24 Hours in USD: 
      0.5591885701135556
    </p>
  </div>
  <p>
    Description: 
    Bitcoin (BTC) is a consensus network that enables a new payment system and a completely digital currency. Powered by its users, it is a peer to peer payment network that requires no central authority to operate. On October 31st, 2008, an individual or group of individuals operating under the pseudonym "Satoshi Nakamoto" published the Bitcoin Whitepaper and described it as: "a purely peer-to-peer version of electronic cash, which would allow online payments to be sent directly from one party to a...
  </p>
</div>
`;
